### PR TITLE
Add comprehensive security review and upgrade plan

### PR DESCRIPTION
**Pull request by dxcSithLord**
This commit provides a complete security analysis and modernization plan for the SQRL GORM authentication store library.

Security Analysis:
- Identified CWE-226: Sensitive data not cleared from memory (Idk, Suk, Vuk keys)
- Identified CWE-200: Sensitive information exposure risks
- Identified CWE-20: Lack of input validation on identity keys
- Documented SQL injection protection (already using parameterized queries)

New Security Implementations:
- Platform-aware secure memory clearing (Unix/Linux and Windows)
- WipeBytes() function to securely zero sensitive data
- ClearIdentity() to wipe all cryptographic fields from SqrlIdentity
- SecureIdentityWrapper for RAII-style automatic cleanup
- Input validation with ValidateIdk() and custom error types

Dependency Upgrade Plan:
- github.com/jinzhu/gorm (deprecated) → gorm.io/gorm v1.31.1
- Added go.mod for modern Go module management
- Target Go 1.23+ (currently supported: Go 1.23, Go 1.24)
- Documented breaking API changes (IsRecordNotFoundError → errors.Is)

CI/CD Pipeline:
- GitHub Actions workflow with lint, test, security, and build jobs
- golangci-lint configuration with security-focused linters
- Makefile for development automation
- Security scanning with gosec and govulncheck
- Test coverage threshold enforcement (70% minimum)
- Multi-platform build verification

Test Coverage:
- Comprehensive test suite for secure memory functions
- Unit tests for WipeBytes, ClearIdentity, ValidateIdk
- Benchmark tests for performance baseline
- Edge case and error condition coverage

Documentation:
- SECURITY_REVIEW_AND_UPGRADE_PLAN.md: Complete security audit
- TODO.md: Actionable checklist with priorities
- Detailed implementation roadmap (4 phases, ~146 hours)

This provides a foundation for securing cryptographic key handling and modernizing the codebase to current Go ecosystem standards.

## Release Notes

* **New Features**
  * Added secure memory wiping for sensitive identity data with platform-specific implementations (Windows and Unix/Linux).
  * Introduced centralized error handling with standardized error definitions.

* **Testing**
  * Added comprehensive test coverage for secure memory operations and identity handling.

* **Chores**
  * Established CI/CD automation with GitHub Actions for linting, security scanning, and multi-platform builds.
  * Added development tooling configuration and build automation via Makefile.
  * Updated module dependencies and Go version configuration.

### CODE REVIEW

**Comment by dxcSithLord** [comment]
@coderabbitai review

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module github.com/sqrldev/server-go-ssp-gormauthstore
+
+go 1.24.0
+
+toolchain go1.24.7
+
+require (
+	github.com/jinzhu/gorm v1.9.16
+	github.com/sqrldev/server-go-ssp v0.0.0-20241212182118-c8230b16b87d
```
Comment (dxcSithLord) [inline go.mod:11]:
```suggestion
	github.com/dxcSithLord/server-go-ssp v0.0.0-20241212182118-c8230b16b87d
```

Comment (dxcSithLord) [reply go.mod:11]: this needs to apply to all references of sqrldev, until the code is pushed back there.
